packages/wdio-browserstack-service/src/logPatcher.ts

```diff
@@ -0,0 +1,52 @@
+import Transport from 'winston-transport'
+// const { consoleHolder } = require('../../helpers/test-observability/constants');
+const consoleHolder: typeof console = Object.assign({}, console)
+
+const LOG_LEVELS = {
+    INFO: 'INFO', ERROR: 'ERROR', DEBUG: 'DEBUG', TRACE: 'TRACE', WARN: 'WARN',
+}
+
+class logPatcher extends Transport {
+    constructor(opts: any) {
+        super(opts)
+    }
+
+    logToTestOps = (level = LOG_LEVELS.INFO, message = ['']) => {
+        // @ts-ignore
+        consoleHolder[level.toLowerCase()](...message)
+        // @ts-ignore
+        process.emit(`bs:addLog:${process.pid}`, {
```

Comment:
Have we verified this for parallel scenario, that if the process object belongs to child or parent process ?

Reply:
Yup, verified for parallel cases too. Our report handler and logPatcher both works in the same process.